File: project/Frontend/src/Services/UserProfileService.ts
import api from '../config/Api';
import {AxiosResponse} from 'axios';
import {UserProfile} from '../types/models/UserProfile.model';
import {UserProfileDTO} from '../types/models/UserProfileDTO.model';

//The user profile service used to use the functionality from the backend for the CRUD methods
const UserProfileService = {
    getUserProfile: async (userProfileID: string): Promise<UserProfile> => {
        const {data} = await api.get<UserProfile>(`/userprofile/${userProfileID}`);
        return data;
    },

    updateUserProfile: (id: string, userProfileDTO: UserProfileDTO) => {
        return api.put(`/userprofile/${id}`, userProfileDTO);
    },

    addUserProfile: (userProfileDTO: UserProfileDTO) => {
        return api.post('/userprofile', userProfileDTO).then((res) => {
            return res.data;
        });
    },

    getAllUserProfiles: (pageSize: number, page: number, sortBy: string, asc: boolean): Promise<AxiosResponse<UserProfile[]>> => {
        return api.get<UserProfile[]>('/userprofile', {
            params: {
                pageSize: pageSize,
                page: page,
                sortBy: sortBy,
                asc: asc
            },
        });
    },

    deleteUserProfile: (id: string) => {
        return api.delete(`/userprofile/${id}`);
    },
};

export default UserProfileService;

File: project/Backend/src/main/java/ch/tbz/testing/domain/userprofile/UserProfileService.java
package ch.tbz.testing.domain.userprofile;

import ch.tbz.testing.core.generic.AbstractService;

import java.util.List;
import java.util.UUID;

public interface UserProfileService extends AbstractService<UserProfile> {
    List<UserProfile> findAllUserProfiles(int page, int pageSize, String sortBy, boolean asc);

    UserProfile findUserProfileById(UUID id);

    UserProfile createUserProfile(UserProfile userProfile);

    UserProfile updateUserProfile(UUID id, UserProfile userProfile);

    void deleteUserProfile(UUID id);
}
